### DISCUSSION TITLE
MQTT for Home Assistant

### DISCUSSION THREAD

**Discussion by jmp2000** [aqualinkd/AqualinkD#272] [Q&A] [answered]
Hello!  Newbie here, trying to get MQTT going.  I've got AqualinkD up and running to my WiFi network on a Pi Zero 2 W in the pool control box, and a long running Home Assistant (HA) on a separate Pi4 in the house.   AqualinkD works great, and the webUI is working perfectly (just connects to the IP of the Pi Zero when on a computer browser in my local network at the house).  I have the Mosquitto Broker on HA, and have the MQTT Integration running.   Having trouble connecting to the Pi Zero W / AqualinkD and sending/receiving any commands through MQTT.  Likely have something configured wrong. 

I'm totally new to MQTT, so I'm sure there is something simple that I've not configured correctly.  Really appreciate anyone pointing out what I've messed up here, so I can keep going up the learning curve!   Thanks a lot for any assistance - AqualinkD is a game changer!!

Here's what is in my Home Assistant / Mosquitto Broker Add-On:
- Start On Boot / Watchdog enabled
- No additional/extra configuration of the Add-On

Here's what is in my Home Assistant / MQTT Integration:
- 1 integration entity, shown as "core-mosquitto"
- Under the MQTT Setting / Reconfigure, it shows the following
- Broker: "core-mosquitto"
- Port: "1883"
- Username: "notmyrealusername"
- Password: "notmyrealpassword"
- In the Publish a packet topic it has "aqualinkd"
- In the Listen to a topic it has "aqualinkd" / format JSON content is not enabled

Here is what is in aqualinkd.conf on the Pi Zero W that is running AqualinkD in the pool control box:
#mqtt stuff
mqtt_address = core-mosquitto:1883
mqtt_user = "notmyrealusername"
mqtt_passwd = "notmyrealpassword"
#mqtt_dz_pub_topic = domoticz/in
#mqtt_dz_sub_topic = domoticz/out
mqtt_aq_topic = aqualinkd

**Comment by sfeakes** [comment, marked as answer]
Everything looks good, only thing that may be an issue is local DNS depending on your network. So rather than using ‘mqtt_address = core-mosquitto:1883’ in AqualinkD.conf use the IP address of the HA machine. ‘ something like mqtt_address = 1111.2222.3333.4444:1883’. 

But before you try that, you should look in the AqualinkD logs to see if it’s spitting out any MQTT errors.

**Reply by jmp2000** [reply]
Great tip!  I think that fixed it up - no errors in the Aqualink logs and things on the HA side with MQTT logs appear to be showing a connection established now.   Fingers crossed !  
So my next step - to get things set up in HA, do I just copy into the configuration.yaml file the various blocks of code from the Wiki, correct (adjusting them to my setup/names/etc)?  Then I can create a 'page' on my Lovelace dashboard to start controlling things...

**Comment by sfeakes** [comment]
For HA setup, you should just be able to use the .yaml example and modify as necessary.   I think there have been some changes to HA over the years, so it may need a few modifications.  I don’t use HA so can’t comment too much on that side.  But there are threads on AqualinkD / HA integration here and on HA forums.
